### PR TITLE
Custom highlighted fields

### PR DESCRIPTION
**Pull request by nastasha-solomon**
Fixes https://github.com/elastic/security-docs/issues/3275.

Previews:

- [Create rule | Configure advanced rule settings (optional)](https://security-docs_3852.docs-preview.app.elstc.co/guide/en/security/master/rules-ui-create.html#rule-ui-advanced-params): Doc'd the new **Custom highlighted fields** setting
- [Add and manage exceptions | Add exceptions to a rule](https://security-docs_3852.docs-preview.app.elstc.co/guide/en/security/master/add-exceptions.html#detection-rule-exceptions): Updated note in step 3.
- [Create rule API | Optional fields for all rule types](https://security-docs_3852.docs-preview.app.elstc.co/guide/en/security/master/rules-api-create.html#opt-fields-all): Added the new param

### CODE REVIEW

File: docs/detections/api/rules/rules-api-create.asciidoc [hunk 1]
```diff
@@ -329,6 +329,10 @@ Required when `actions` are used to send notifications.
 
 |version |Integer |The rule's version number. Defaults to `1`.
 
+|investigation_fields |String a| Specify additional highlighted fields for the alert:
```
Comment (yctercero) [inline docs/detections/api/rules/rules-api-create.asciidoc:332]: What do you think of:

```
Specify additional highlighted fields for select alert investigation flows:
```

Comment (nastasha-solomon) [reply docs/detections/api/rules/rules-api-create.asciidoc:332]: Are "select alert investigation flows" personalized investigation flows? Or are we saying that customized highlighted fields only select _a few_ investigation flows? If it's the former, I can modify the description to:

Specify highlighted fields for personalized alert investigation flows.

Comment (yctercero) [reply docs/detections/api/rules/rules-api-create.asciidoc:332]: That's a good question. With "select alert investigation flows" I was trying to reference the fact that these fields are used for more than just the alert details > highlighted fields section (like alert details > prevelance and exceptions). But maybe that's just more confusing. 

And ++ to adding that field description to the create rule docs. Thanks for catching that!

Comment (nastasha-solomon) [reply docs/detections/api/rules/rules-api-create.asciidoc:332]: Ok, cool. So just to wrap this up, do you think the description is good as is, or should I change it to: 
```suggestion
|investigation_fields |String a| Specify highlighted fields for personalized alert investigation flows:
```

File: docs/detections/rules-ui-create.asciidoc [hunk 2]
```diff
@@ -402,6 +402,8 @@ the rule. For example, links to background information.
 .. *False positive examples* (optional): List of common scenarios that may produce
 false-positive alerts.
 .. *MITRE ATT&CK^TM^ threats* (optional): Add relevant https://attack.mitre.org/[MITRE] framework tactics, techniques, and subtechniques.
+.. *Custom highlighted fields* (optional): Specify additional fields to display in the [investigation-section,Highlighted fields] section of the alert details flyout.
```
Comment (nastasha-solomon) [inline docs/detections/rules-ui-create.asciidoc:405]:
```suggestion
.. *Custom highlighted fields* (optional): Specify highlighted fields for personalized alert investigation flows. Fields with data are added to the [investigation-section,Highlighted fields] section within the alert details flyout. You can also find custom highlighted fields in the About section of the rule details page.
+
NOTE: There's no maximum limit to the number of custom highlighted fields you can add.  
```

File: docs/detections/rules-ui-create.asciidoc [hunk 3]
```diff
@@ -402,6 +402,10 @@ the rule. For example, links to background information.
 .. *False positive examples* (optional): List of common scenarios that may produce
 false-positive alerts.
 .. *MITRE ATT&CK^TM^ threats* (optional): Add relevant https://attack.mitre.org/[MITRE] framework tactics, techniques, and subtechniques.
+.. *Custom highlighted fields* (optional): Specify highlighted fields for personalized alert investigation flows. Fields with data are added to the [investigation-section,Highlighted fields] section within the alert details flyout. You can also find custom highlighted fields in the About section of the rule details page.
++
+NOTE: There's no maximum limit to the number of custom highlighted fields you can add.
```
Comment (nastasha-solomon) [inline docs/detections/rules-ui-create.asciidoc:407]:
```suggestion
NOTE: There's no limit to the number of custom highlighted fields you can add.  
```

File: docs/detections/add-exceptions.asciidoc [hunk 4]
```diff
@@ -62,7 +62,7 @@ image::images/rule-exception-tab.png[Detail of rule exceptions tab]
 . In the *Add rule exception* flyout, name the exception. 
 . Add conditions that define the exception. When the exception's query evaluates to `true`, rules don't generate alerts even when their criteria are met.
 +
-NOTE: When you create a new exception from an alert, exception conditions are auto-populated with relevant alert data. A comment describing this action is also automatically added to the **Add comments** section.
+NOTE: When you create a new exception from an alert, exception conditions are auto-populated with relevant alert data, including data from custom highlighted fields. A comment describing this action is also automatically added to the **Add comments** section.
```
Comment (yctercero) [inline docs/detections/add-exceptions.asciidoc:65]:
```suggestion
NOTE: When you create a new exception from an alert, exception conditions are auto-populated with relevant alert data, prioritizing data matching any custom highlighted fields. A comment describing this action is also automatically added to the **Add comments** section.
```

Comment (nastasha-solomon) [reply docs/detections/add-exceptions.asciidoc:65]: Thanks for the edits, @yctercero :) I made some minor tweaks - let me know if they work:

```suggestion
NOTE: When you create a new exception from an alert, exception conditions are auto-populated with relevant alert data. Data from custom highlighted fields are listed first. A comment that describes the auto-generated exceptions is also added to the **Add comments** section.
```

Comment (yctercero) [reply docs/detections/add-exceptions.asciidoc:65]: That sounds great! Thanks so much.

**Review by yctercero — APPROVED** [review]

File: docs/detections/rules-ui-create.asciidoc [hunk 5]
```diff
@@ -316,6 +316,10 @@ the rule. For example, links to background information.
 .. *False positive examples* (optional): List of common scenarios that may produce
 false-positive alerts.
 .. *MITRE ATT&CK^TM^ threats* (optional): Add relevant https://attack.mitre.org/[MITRE] framework tactics, techniques, and subtechniques.
+.. *Custom highlighted fields* (optional): Specify highlighted fields for personalized alert investigation flows. Fields with data are added to the [investigation-section,Highlighted fields] section within the alert details flyout. You can also find custom highlighted fields in the About section of the rule details page.
```
Comment (natasha-moore-elastic) [inline docs/detections/rules-ui-create.asciidoc:319]: Is "Fields with data are added..." meant to imply that fields with no values wouldn't be added to these sections? If so, would it be beneficial to state this explicitly?

File: docs/detections/add-exceptions.asciidoc [hunk 6]
```diff
@@ -62,7 +62,7 @@ image::images/rule-exception-tab.png[Detail of rule exceptions tab]
 . In the *Add rule exception* flyout, name the exception. 
 . Add conditions that define the exception. When the exception's query evaluates to `true`, rules don't generate alerts even when their criteria are met.
 +
-NOTE: When you create a new exception from an alert, exception conditions are auto-populated with relevant alert data. A comment describing this action is also automatically added to the **Add comments** section.
+NOTE: When you create a new exception from an alert, exception conditions are auto-populated with relevant alert data. Data from custom highlighted fields is listed first. A comment that describes the auto-generated exceptions is also added to the **Add comments** section.
```
Comment (natasha-moore-elastic) [inline docs/detections/add-exceptions.asciidoc:65]:
```suggestion
NOTE: When you create a new exception from an alert, exception conditions are auto-populated with relevant alert data. Data from custom highlighted fields is listed first. A comment that describes the auto-generated exception conditions is also added to the **Add comments** section.
```

**Review by natasha-moore-elastic — COMMENTED** [review]
Left a couple of questions, otherwise LGTM!

File: docs/detections/api/rules/rules-api-create.asciidoc [hunk 7]
```diff
@@ -329,6 +329,10 @@ Required when `actions` are used to send notifications.
 
 |version |Integer |The rule's version number. Defaults to `1`.
 
+|investigation_fields |String a| Specify highlighted fields for personalized alert investigation flows:
```
[the remark this answers is not in the document]
Comment (nastasha-solomon) [reply docs/detections/api/rules/rules-api-create.asciidoc:332]: Actually, I think this is right. I was thinking about the value input type, not the field type when I wrote this.

**Review by benironside — APPROVED** [review]
LGTM

File: docs/detections/rules-ui-create.asciidoc [hunk 8]
```diff
@@ -316,6 +316,10 @@ the rule. For example, links to background information.
 .. *False positive examples* (optional): List of common scenarios that may produce
 false-positive alerts.
 .. *MITRE ATT&CK^TM^ threats* (optional): Add relevant https://attack.mitre.org/[MITRE] framework tactics, techniques, and subtechniques.
+.. *Custom highlighted fields* (optional): Specify highlighted fields for personalized alert investigation flows. Fields with values are added to the [investigation-section,Highlighted fields] section within the alert details flyout. Fields without values aren't added. After you create the rule, you can find all custom highlighted fields in the About section of the rule details page.
```
Comment (nastasha-solomon) [inline docs/detections/rules-ui-create.asciidoc:319]:
```suggestion
.. *Custom highlighted fields* (optional): Specify highlighted fields for personalized alert investigation flows. Fields with values are added to the <<investigation-section,Highlighted fields>> section within the alert details flyout. Fields without values aren't added. After you create the rule, you can find all custom highlighted fields in the About section of the rule details page.
```

File: docs/detections/rules-ui-create.asciidoc [hunk 9]
```diff
@@ -316,6 +316,10 @@ the rule. For example, links to background information.
...
++
+NOTE: There's no limit to the number of custom highlighted fields you can add.  
+//Will change brackets to double angled brackets once merge PR #3816
```
Comment (nastasha-solomon) [inline docs/detections/rules-ui-create.asciidoc:322]: Removing comment because merged PR #3816.
```suggestion
```

**Pull request merged**